src/main/java/com/microsoft/jenkins/containeragents/aci/AciCloud.java

```diff
@@ -177,8 +177,22 @@ public AciContainerTemplate getFirstTemplate(Label label) {
     public void addIpEnv(AciAgent agent) throws Exception {
         AzureResourceManager azureResourceManager = getAzureClient();
 
-        String ip = azureResourceManager.containerGroups()
-                .getByResourceGroup(resourceGroup, agent.getNodeName()).ipAddress();
+        // Workaround for https://github.com/Azure/azure-sdk-for-java/issues/27083
+        String ip = null;
+        boolean nullIsThrown;
+        do {
+            try {
+                ip = azureResourceManager.containerGroups()
```

Comment:
could we check the ip address for null instead and then retry rather than handling a null pointer?
is it possible the IP hasn't been allocated yet? seems quite weird though

Reply:
```text
java.lang.NullPointerException
        at com.azure.resourcemanager.containerinstance.implementation.ContainerGroupImpl.initializeChildrenFromInner(ContainerGroupImpl.java:217)

```